File: Diagrammatize.Design.Pattern/code/python/Observer/A2/GraphObserver.py
from Observer import Observer
import time

class GraphObserver(Observer):
    def update(self, generator):
        print("GraphObserver:", end='')
        count = generator.getNumber()
        for i in range(0, count):
            print('*', end='')
        print('')
        time.sleep(0.1)

File: Diagrammatize.Design.Patterns/code/cxx/Observer/A1/GraphObserver.cpp
#include "GraphObserver.hpp"
#include "NumberGenerator.hpp"
#include <iostream>
#include <thread>
#include <chrono>

void GraphObserver::update(NumberGenerator* generator) {
    std::cout << "GraphObserver:";
    int count = generator->getNumber();
    for (int i = 0; i < count; i++) {
        std::cout << "*";
    }
    std::cout << std::endl;
    std::this_thread::sleep_for(std::chrono::milliseconds(100));
}
